### ISSUE TITLE
Type inference not working on variable from module global variable in library.

### ISSUE THREAD

**Issue by Rawk** [microsoft/pyright#1559]
**Describe the bug**
Type inference not working on variable imported from module global variable in library.

**To Reproduce**
Example using logger from FastAPI.

pyrightconfig.json:
```json
{ "typeCheckingMode": "strict" }
```

my_app.main.py:
```python
import fastapi.logger
fastapi.logger.logger.debug("sdfsdf")
# Error: Type of "logger" is unknown
# Error: Type of "debug" is unknown 

# Importing project file with the same code works as expected
import my_app.logger
my_app.logger.logger.debug("sdfsdf")
```

venv/lib/python3.8/site-packages/fastapi/logger.py AND my_app.logger.py:
```python
import logging
logger = logging.getLogger("fastapi")
```

pyright output:
```
$ pyright --version
pyright 1.1.116
$ pyright --verbose src
Loading configuration file at /home/---/projects/test2/pyrightconfig.json
Assuming Python version 3.8
Assuming Python platform Linux
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding .git
Search paths found for configured python interpreter:
  /usr/lib/python3.8
  /usr/lib/python3.8/lib-dynload
  /home/---/projects/test2/venv/lib/python3.8/site-packages
When attempting to get search paths from python interpreter:
  Executing interpreter: 'python3'
  Skipping '/usr/lib/python38.zip' because it is not a valid directory
  Received 3 paths from interpreter
    /usr/lib/python3.8
    /usr/lib/python3.8/lib-dynload
    /home/---/projects/test2/venv/lib/python3.8/site-packages
stubPath /home/---/projects/test2/typings is not a valid directory.
Searching for source files
Found 2 source files
/home/---/projects/test2/src/main.py
  5:1 - error: Type of "logger" is unknown (reportUnknownMemberType)
  5:1 - error: Type of "debug" is unknown (reportUnknownMemberType)
2 errors, 0 warnings, 0 infos 
Completed in 0.681sec
```

**Expected behavior**
The type of `fastapi.logger.logger` to be infered as `logging.Logger` from return type of `logging.getLogger(..)`.

**VS Code extension or command-line**
Pyright CLI version 1.1.116.
Also happens with VS Code Pylance plugin version 2021.2.4. I think it uses Pyright 1.1.114.

**Comment by erictraut** [comment]
This is intended behavior. The `fastapi` library is a "py.typed" package. As such, it is expected to clearly specify which symbols comprise its public interface contract and provide explicit type annotations for these symbols. For more details about how pyright handles py.typed libraries, refer to [this documentation](https://github.com/microsoft/pyright/blob/master/docs/typed-libraries.md).

There are a couple of reasons we don't attempt any form of type inference for "py.typed" libraries. First, type inference rules are not standardized across type checkers, and it's important for behavior to be consistent across type checkers. Second, type inference can be very expensive, and speed is important when resolving symbol types for interactive editing features (completion suggestions, signature help, etc.)

In your particular example, the `fastapi.logger` submodule does not provide an annotation for the symbol `logger`, so its type is Unknown. Please report it to the maintainers of the fastapi library so they can address this. 

The CLI version of pyright has a "--verifytypes" feature that library maintainers can run on their library to find missing type annotations. Here is the output for the latest published version of fastapi (0.63.0):
```
Public symbols: 676
  Symbols with unknown type: 343
  Functions with missing docstring: 92
  Functions with missing default param: 0
  Classes with missing docstring: 80
Type completeness score: 49.3%
```

**Comment by erictraut** [comment]
As a temporary workaround, you can delete the "py.typed" file from the fastapi package after it is installed. Pyright will then revert back to its non-py.typed behavior and attempt to infer types if `useLibraryCodeForTypes` is enabled.

**Comment by Rawk** [comment]
Great answer. Thanks!
